cgnsutilities/cgnsutilities.py

```diff
@@ -2942,43 +2948,75 @@ def convertPlot3d(plot3dFile, cgnsFile):
     libcgns_utils.utils.convertplot3d(plot3dFile, cgnsFile)
 
 
-def mirrorGrid(grid, axis, tol, useOldNames=False):
+def mirrorGrid(grid, axis, tol, useOldNames=True):
     """Method that takes a grid and mirrors about the axis. Boundary
     condition information is retained if possible"""
 
     # First make sure the grid is face matched:
     grid.split([])
 
-    # Now copy original blocks
+    # create the new grid object
     newGrid = Grid()
 
-    for blk in grid.blocks:
-        blk.removeSymBCs()
-        blk.B2Bs = []
-        newGrid.addBlock(blk)
+    # rename the new grid if asked for
+    if useOldNames:
+        newGrid.name = grid.name
 
-        if useOldNames:
-            # Add the current block name to the new grid
-            blk.name = blk.name.split(".")[0]
+    # Now copy original blocks
+    for blk in grid.blocks:
+        new_blk = copy.deepcopy(blk)
+        new_blk.removeSymBCs()
+        new_blk.B2Bs = []
+        newGrid.addBlock(new_blk)
 
-        mirrorBlk = copy.deepcopy(blk)
+        mirrorBlk = copy.deepcopy(new_blk)
         mirrorBlk.flip(axis)
-        newGrid.addBlock(mirrorBlk)
-
         if useOldNames:
-            # Add the new mirrored block name to the new grid
-            newGrid.blocks[-1].name = blk.name.split(".")[0] + "_mirror"
-
-            print(f"Mirroring block: {blk.name} to {newGrid.blocks[-1].name}")
+            # overwrite the name of the mirror block
+            mirrorBlk.name = blk.name.split(".")[0] + "_mirror." + blk.name.split(".")[-1]
+        newGrid.addBlock(mirrorBlk)
 
     # Now rename the blocks and redo-connectivity
-    newGrid.renameBlocks(useOldNames=useOldNames)
+    if not useOldNames:
+        newGrid.renameBlocks()
     newGrid.renameBCs()
     newGrid.connect(tol)
 
     return newGrid
 
 
+def mirrorGridSurface(grid, axis, useOldNames=True):
```

Comment:
This mostly duplicates `mirrorGrid`. How about a kwarg to  `mirrorGrid` like `surface=False`? `surface=True` would skip the lines that we don't want for surface meshes.

Reply:
yeah I was thinking about it but was lazy... will do

Reply:
done